man/systemd.socket.xml

```diff
@@ -429,6 +429,16 @@
         false, in read-only mode. Defaults to false.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>ClearSocket=</varname></term>
```

Comment:
Hmm, is "Clear" the right name for this? usually something like this is called "Flush", no?

Also the "Socket" is kinda redundant here, since this is a *socket* unit anyway we are talking about here... Maybe just call this "Flush"?

Reply:
> Hmm, can you elaborate what the usecase for this is? I mean, I am not against this, it kinda makes sense to me, but I'd be curious what your precise usecase is for this?

I'm filing this on behalf of a Red Hat customer. The customer cannot give much details because it's part of a banking application. Nonetheless below is what I understood:
The use case is with Accept=no only. Upon the socket triggering the service due to some incoming data, in case the service fails or even exits properly, their application requires the socket to not respin the service automatically even if there is still pending data on the socket. They only want the socket to spin the service again on newer incoming data.
I understand this may look kinda odd since systemd will not be able to distinguish between new data and existing pending ones on the socket at the time of the service failure.
My guess is the use case is some kind of batch running that should abort upon service failure. By flushing the data the batch will probably fail as well.

Reply:
> Hmm, is "Clear" the right name for this? usually something like this is called "Flush", no?
> 
> Also the "Socket" is kinda redundant here, since this is a _socket_ unit anyway we are talking about here... Maybe just call this "Flush"?

Let's go for Flush then.

Reply:
hmm the customer must be able to explain in generic terms what he wants to do with this, no?

Reply:
> Hmm, is "Clear" the right name for this? usually something like this is called "Flush", no?
> 
> Also the "Socket" is kinda redundant here, since this is a _socket_ unit anyway we are talking about here... Maybe just call this "Flush"?

The customer proposes "FlushOnServiceExit" which is more human-friendly than "Flush".
@poettering What do you think?

Reply:
> The customer proposes "FlushOnServiceExit" which is more human-friendly than "Flush".
> @poettering What do you think?

Well, the code actually flushes whenever we enter the listening state, i.e. even right after we created/bound the socket, hence calling it FlushOnServiceExit() is a tiny bit misleading... Of course, the fact that it flushes right after creating/binding the socket is nothing we should advertise, but it still means we should keep the name generic.

Reply:
but maybe @keszybz has some input on naming this?

Reply:
`FlushPendingConnections=`? (Based on the description below.)
(Flush= is cryptic, and  FlushOnServiceExit= is misleading, not just a bit...)